content/doc/book/using/slave-master-access-control.adoc

```diff
@@ -0,0 +1,484 @@
+your master, then it is safe to leave this subsystem off. The following
+are examples of deployments that fit this profile:
+
+* You have set up your Jenkins master and all the agents by yourself
+* All the users of your Jenkins have the same access to all the jobs
+
+For this subsystem to function, it needs cooperation from plugins. And
+until plugins get updated to work in this new environment, this may
+cause seemingly legitimate operations to fail, resulting in broken
+builds and other failures.
+
+This Wiki page is intended to be the one-stop shop for those who are
+experiencing problems due to this subsystem.
+
+== My Jenkins is broken, help!
+
+If you see an exception like the following in console output, logs,
+etc., then you are affected by this access control mechanism:
+
+....
+java.lang.SecurityException: agent may not create /var/lib/jenkins/foo/bar/...
+See http://jenkins-ci.org/security-144 for more details
+
+java.lang.SecurityException: Sending org.jenkinsci.plugins.gitclient.CliGitAPIImpl$GetPrivateKeys from agent to master is prohibited.
+See http://jenkins-ci.org/security-144 for more details
+....
+
+The first message indicates that an agent tried to access a file on the
+master, and Jenkins has blocked that. It is possible that this indicates
+an active attack, but it is also possible that plugins that haven't yet
+upgraded to work with SECURITY-144 is doing something legitimate.
+
+The second message indicates that an agent has asked the master to
+execute a command, and Jenkins has blocked that because the command
+isn't classified by plugin developers whether it is safe to be executed
+from the agent.
+
+To resolve this problem, you have the following options:
+
+* https://wiki.jenkins.io/display/JENKINS//Slave+To+Master+Access+Control#SlaveToMasterAccessControl-update-plugins[Check
+if the updates to plugins are available]
+* https://wiki.jenkins.io/display/JENKINS//Slave+To+Master+Access+Control#SlaveToMasterAccessControl-kill-switch[Disable
+the access control mechanism altogether]
+* https://wiki.jenkins.io/display/JENKINS//Slave+To+Master+Access+Control#SlaveToMasterAccessControl-whitelist[Whitelist
+the access]
+
+[[SlaveToMasterAccessControl-Checkiftheupdatestopluginsareavailable]]
+=== Check if the updates to plugins are available
+
+During the testing of this feature, the following plugins and their
+versions have been known to interfere with this subsystem. If you are
+using one of those, upgrading those plugins will resolve the problem.
+This section will be updated as we discover affected plugins.
+
+[[SlaveToMasterAccessControl-CredentialsPlugin]]
+==== https://wiki.jenkins.io/display/JENKINS/Credentials+Plugin[Credentials Plugin]
+
+This plugin should be upgraded to 1.17. Earlier versions have command
+implementations that are vulnerable to malicious agents.
+
+[[SlaveToMasterAccessControl-SSHCredentialsPlugin]]
+==== https://wiki.jenkins.io/display/JENKINS/SSH+Credentials+Plugin[SSH Credentials Plugin]
```

Comment:
```suggestion
==== https://plugins.jenkins.io/ssh-credentials/[SSH Credentials Plugin]
```

Reply:
I think it may be better if we drop this section completely.  The plugins referenced are now far beyond those version numbers and Jenkins is now far beyond the base versions described here.

Reply:
noted